### PR TITLE
Update qft.py

### PR DESCRIPTION
**Pull request by JakeKitchen**
**Context:**

The original implementation recalculated the length of the `shifts` list inside the loop during each iteration, which introduces unnecessary repetition.

---

**Description of the Change:**

moving the computation of `shift_len = len(shifts)` outside the loop in the `compute_decomposition` function., it is now computed once before the loop starts and reused.

---

**Benefits:**

- **Performance Improvement:** The length of the `shifts` list is now only computed once, reducing unnecessary calculations and slightly improving performance, especially for cases with a large number of wires or repeated calls.
- **Readability:** The code is clearer as the intent to calculate the length only once is explicitly shown.
- **Maintainability:** It simplifies the logic by ensuring that `shift_len` is calculated only once, reducing the risk of future errors or bugs related to recalculating the list length.

---

**Possible drawbacks:**

- None

---

**Related GitHub Issues:**

- N/A

### CODE REVIEW

**Review by JerryChen97 — APPROVED** [review]
LGTM! Thank you so much Jake

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -406,6 +409,7 @@ Pietropaolo Frisoni,
 Emiliano Godinez,
 Austin Huang,
 Korbinian Kottmann,
+Jacob Kitchen,
```
Comment (andrijapau) [inline doc/releases/changelog-dev.md:412]:
```suggestion
Jacob Kitchen,
Korbinian Kottmann,
```

Just a minor suggestion for alphabetical ordering. 😄

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -189,6 +189,9 @@
 * A more sensible error message is raised from a `RecursionError` encountered when accessing properties and methods of a nested `CompositeOp` or `SProd`.
   [(#6375)](https://github.com/PennyLaneAI/pennylane/pull/6375)
 
+* Moved the calculation of `shift_len = len(shifts)` outside of a loop in the `compute_decomposition` function, reducing redundant recalculations and improving performance.
```
Comment (andrijapau) [inline doc/releases/changelog-dev.md:192]:
```suggestion
* Moved the calculation of `shift_len = len(shifts)` outside of a loop in the `QFT.compute_decomposition` static method, reducing redundant recalculations and improving performance.
```

Just another minor suggestion - will help in the future if we ever need to re-read the changelog. 😄

Comment (JakeKitchen) [reply doc/releases/changelog-dev.md]: LGTM!

**Comment by JakeKitchen** [comment]
Not sure why the codecov dropped "Failing after 1s — 99.38% (-0.32%) compared to 2053da9" changes are out of scope from the changes made.

**Comment by albi3ro** [comment]
> Not sure why the codecov dropped "Failing after 1s — 99.38% (-0.32%) compared to [2053da9](https://github.com/PennyLaneAI/pennylane/commit/2053da963a36554f85c80e276b48c9369b87b116)" difference in the codecov is out of scope from the changes made. Assuming its a false positive or CI related problem

We've just been dealing with some code cov issues lately due to a large amount of deprecated code that was difficult to continue testing. So I overrode.

**Pull request merged**